### PR TITLE
Add a vector dimension

### PR DESCRIPTION
**Pull request by m-mohr**
A WIP draft for a new dimension of type vector.

### CODE REVIEW

**Comment by mkadunc** [comment]
> The other question which representations we allow in metadata. Right now it supports:
> - GeoJSON (full set right now, so includes Features, but we may want to remove FeatureCollections though, see above)
> - WKT for Geometries
> - IDs

Is it still a vector cube if it has IDs as labels of the vector dimension? For all uses in processes, such a cube would be just a normal `data-cube`; what use is calling such a dimension `vector`?

What is the benefit of supporting both GeoJSON and WKT? I'd go with just GeoJSON.

**Comment by mkadunc** [comment]
> Should we add a vector type field to indicate whether e.g. all geometries are points?

yes

**Comment by m-mohr** [comment]
Added `geometry_types` and `axes` as new fields.

File: README.md [hunk 1]
```diff
@@ -85,13 +85,27 @@ A temporal dimension based on the ISO 8601 standard. The temporal reference syst
...
+| description      | string         | Detailed multi-line description to explain the dimension. [CommonMark 0.29](http://commonmark.org/) syntax MAY be used for rich text representation. |
+| bbox             | \[number]      | **REQUIRED.** A single bounding box of the geometries as defined for [STAC Collections](https://github.com/radiantearth/stac-spec/blob/master/collection-spec/collection-spec.md#spatial-extent-object), but not nested. |
+| values           | \[string\|[GeoJSON Object](https://tools.ietf.org/html/rfc7946)] | Optionally, a representation of the vectors. This can be a list of WKT strings, GeoJSON objects or identifiers. |
+| geometry_types   | \[[GeoJSON Types](https://www.rfc-editor.org/rfc/rfc7946#section-1.4)] | A set of geometry types. If not present, mixed geometry types must be assumed. |
+| reference_system | string\|number\|object | The spatial reference system for the data, specified as [numerical EPSG code](http://www.epsg-registry.org/), [WKT2 (ISO 19162) string](http://docs.opengeospatial.org/is/18-010r7/18-010r7.html) or [PROJJSON object](https://proj.org/specifications/projjson.html). Defaults to EPSG code 4326. |
+
```
Comment (edzer) [inline README.md:101]: Instead of a number, it would be better to use identifiers such as `EPSG:4326`, as this allows for other authorities than EPSG (e.g., ESRI, OGC). A default that is better than `EPSG:4326` is `OGC:CRS84`; it points to the same datum (ensemble?) but makes explicit that coordinates are pairs of (longitude,latitude), where `EPSG:4326` _suggests_ that coordinates are pairs of (latitude,longitude) which in practice they almost always aren't. Using `OGC:CRS84` leaves that ambiguity behind us.

File: json-schema/schema.json [hunk 2]
```diff
@@ -372,6 +371,86 @@
...
+          "items": {
+            "type": "string",
+            "enum": [
+              "Point",
+              "MultiPoint",
+              "LineString",
+              "MultiLineString",
+              "Polygon",
+              "MultiPolygon",
+              "GeometryCollection"
```
Comment (edzer) [inline json-schema/schema.json:445]: In other places you used `Multi Point`, `Multi LineString`, `Multi Polygon`; I guess this one is the correct one (standards compliant)?

Comment (m-mohr) [reply json-schema/schema.json:445]: Where are these other places? I can't find any of these used with a space in-between.

File: README.md (hunk 1 above)
Comment (m-mohr) [reply README.md:101]: Yeah, that ship has sailed in STAC. We either use EPSG code or PROJJSON or WKT2 for now...
We can say we default to another CRS, but we can then only encode it in PROJJSON or WKT2.

File: json-schema/schema.json (hunk 2 above)
Comment (edzer) [reply json-schema/schema.json:445]: https://github.com/Open-EO/openeo.org/pull/59

Comment (m-mohr) [reply json-schema/schema.json:445]: Ah, okay, these ones here are used in the STAC extension (and GeoJSON etc) for M2M, the others are just textual descriptions for humans.

File: README.md [hunk 3]
```diff
@@ -85,13 +85,27 @@ A temporal dimension based on the ISO 8601 standard. The temporal reference syst
 | values     | \[string]       | If the dimension consists of an ordered list of specific values they can be listed here. The dates and/or times must be strings compliant to [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601). |
 | step       | string\|null    | The space between the temporal instances as [ISO 8601 duration](https://en.wikipedia.org/wiki/ISO_8601#Durations), e.g. `P1D`. Use `null` for irregularly spaced steps. |
 
+### Vector Dimension Object
```
Comment (m-mohr) [inline README.md:88]:
```suggestion
### Geometry Dimension Object
```

**Comment by m-mohr** [comment]
@soxofaan @mkadunc @aljacob @clausmichele @jdries @dthiex @LukeWeidenwalker @pierocampa This is also ready for final review.

**Review by edzer — APPROVED** [review]

**Review by mkadunc — APPROVED** [review]

**Review by aljacob — APPROVED** [review]
As discussed yesterday, looks all good to me now, thanks for the hard work!!

**Comment by jdries** [comment]
nice work!

File: examples/vector.json [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  },
+  "bbox": [
+    -122.51575469970703,
+    37.50373410549169,
+    -122.10994720458984,
+    37.747507619675595
+  ],
+  "properties": {
+    "title": "Multi-dimensional vector data cube in a STAC Item.",
+    "datetime": "2016-05-03T13:21:30.040Z",
```
Comment (clausmichele) [inline examples/vector.json:117]: How is this related with the values in cube:dimensions -> "t" -> "temporal" ?

**Review by clausmichele — APPROVED** [review]

File: examples/vector.json (hunk 4 above)
Comment (m-mohr) [reply examples/vector.json]: Good point, I updated the example to include start and end datetimes.
